main/src/cgeo/geocaching/maps/PositionAndScaleOverlay.java

```diff
@@ -16,22 +19,40 @@
 
     PositionDrawer positionDrawer = null;
     ScaleDrawer scaleDrawer = null;
+    DirectionDrawer directionDrawer = null;
+    DistanceDrawer distanceDrawer = null;
 
-    public PositionAndScaleOverlay(OverlayImpl ovlImpl) {
+    public PositionAndScaleOverlay(final OverlayImpl ovlImpl, final MapViewImpl mapView, final Geopoint coords, final String geocode) {
         this.ovlImpl = ovlImpl;
         positionDrawer = new PositionDrawer();
         scaleDrawer = new ScaleDrawer();
+
+        if (coords != null) {
+            directionDrawer = new DirectionDrawer(coords);
+            distanceDrawer = new DistanceDrawer(mapView, coords);
+        } else if (geocode != null) {
+            final Viewport bounds = DataStore.getBounds(geocode);
+            if (bounds != null) {
+                directionDrawer = new DirectionDrawer(bounds.center);
+                distanceDrawer = new DistanceDrawer(mapView, bounds.center);
+            }
+        }
     }
 
-    public void setCoordinates(Location coordinatesIn) {
+    public void setCoordinates(final Location coordinatesIn) {
         positionDrawer.setCoordinates(coordinatesIn);
+        if (directionDrawer != null) {
```

Comment:
Assume that `distanceDrawer` is also `!= null`.

Well, in this case it should work, but there must be better ways to draw this stuff if necessary.